# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerChild.h
# Commit: 77eda17d0795
# Full Hash: 77eda17d0795e32bd6bf224832b922d8e57aedca
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-15 21:21:06
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r 358f0cf9a8d8 -r 77eda17d0795 dom/workers/remoteworkers/RemoteWorkerDebuggerChild.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerChild.h	Tue Apr 15 09:47:38 2025 +0000
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_RemoteWorkerDebuggerChild_h
+#define mozilla_dom_RemoteWorkerDebuggerChild_h
+
+#include "mozilla/dom/PRemoteWorkerDebuggerChild.h"
+#include "mozilla/RefPtr.h"
+
+using mozilla::ipc::IPCResult;
+
+namespace mozilla::dom {
+
+class WorkerPrivate;
+
+class RemoteWorkerDebuggerChild final : public PRemoteWorkerDebuggerChild {
+  friend class PRemoteWorkerDebuggerChild;
+
+ public:
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(RemoteWorkerDebuggerChild, final)
+
+  explicit RemoteWorkerDebuggerChild(WorkerPrivate* aWorkerPrivate);
+
+  mozilla::ipc::IPCResult RecvInitialize(const nsString& aURL);
+  mozilla::ipc::IPCResult RecvPostMessage(const nsString& aMessage);
+  mozilla::ipc::IPCResult RecvSetDebuggerReady(const bool& aReady);
+
+ private:
+  ~RemoteWorkerDebuggerChild();
+
+  RefPtr<WorkerPrivate> mWorkerPrivate;
+};
+
+}  // namespace mozilla::dom
+
+#endif  // mozilla_dom_RemoteWorkerDebuggerChild_h